resource-manager/src/main/java/org/ihtsdo/otf/resourcemanager/ResourceManager.java

```diff
@@ -1,113 +1,409 @@
-
-	private void writeCheck() {
-		if (resourceConfiguration.isReadonly()) {
-			throw new UnsupportedOperationException("Can not write resources in this read-only resource manager.");
-		}
-	}
-
-	private String getFullPath(String relativePath) {
-		if (resourceConfiguration.isUseCloud()) {
-			return "s3://" + resourceConfiguration.getCloud().getBucketName() + "/" +
-					getPathAndRelative(resourceConfiguration.getCloud().getPath(), relativePath);
-		}
-		return getPathAndRelative(resourceConfiguration.getLocal().getPath(), relativePath);
-	}
-
-	// Make sure we have a single slash between path and relative path
-	private String getPathAndRelative(String path, String relativePath) {
-		if (!path.isEmpty() && !path.endsWith("/")) {
-			path += "/";
-		}
-		if (relativePath.startsWith("/")) {
-			relativePath = relativePath.substring(1);
-		}
-		return path + relativePath;
-	}
+    private final ResourceConfiguration resourceConfiguration;
+    private final ResourceLoader resourceLoader;
+    private AmazonS3 amazonS3;
+
+    /**
+     * Creates the {@link ResourceManager} with the corresponding <code>resourceConfiguration</code>
+     * and <code>cloudResourceLoader</code>. {@code NullPointerException} is thrown if
+     * either of the parameters are {@code null}.
+     *
+     * @param resourceConfiguration Used to get the resource configuration which details
+     *                              the local/cloud setting.
+     * @param cloudResourceLoader   The resource loader associated to the cloud
+     *                              service.
+     */
+    public ResourceManager(final ResourceConfiguration resourceConfiguration,
+                           final ResourceLoader cloudResourceLoader) throws NullPointerException {
+        this.resourceConfiguration = Objects.requireNonNull(resourceConfiguration);
+        if (resourceConfiguration.isUseCloud()) {
+            this.resourceLoader = checkS3Connection(Objects.requireNonNull(cloudResourceLoader));
+            this.amazonS3 = buildAmazonS3Client(resourceConfiguration);
+        } else {
+            this.resourceLoader = new FileSystemResourceLoader();
+        }
+    }
+
+    /**
+     * Builds the {@link AmazonS3} instance so that delete and move operations can occur on
+     * objects inside S3.
+     *
+     * @param resourceConfiguration Which contains the configuration settings for the access
+     *                              and secret key.
+     * @return {@link AmazonS3} instance so that delete and movement operations can occur on
+     * objects inside S3.
+     */
+    private AmazonS3 buildAmazonS3Client(final ResourceConfiguration resourceConfiguration) {
+        final Cloud cloud = resourceConfiguration.getCloud();
+        return AmazonS3ClientBuilder.standard()
+                                    .withRegion(cloud.getRegion())
+                                    .withCredentials(
+                                            new AWSStaticCredentialsProvider(
```

Comment:
You should be able to replace this with something that gets the creds from the environment vars.

Reply:
Fixed.